eslint.config.mjs

```diff
@@ -0,0 +1,90 @@
+import tsStylistic from '@stylistic/eslint-plugin-ts';
+import prettier from 'eslint-config-prettier';
+import tsEslint from 'typescript-eslint';
+
+import apify from '@apify/eslint-config/ts';
+
+export default [
+	{
+		ignores: [
+			'**/dist',
+			'node_modules',
+			'coverage',
+			'website/{build,.docusaurus}',
+			'**/*.d.ts',
+			'test/tmp/**/*',
+			'.yarn/**/*',
+		],
+	},
+	...apify,
+	prettier,
+	{
+		languageOptions: {
+			parser: tsEslint.parser,
+			parserOptions: {
+				project: 'tsconfig.eslint.json',
+			},
+		},
+	},
+	{
+		plugins: {
+			'@typescript-eslint': tsEslint.plugin,
+			'@stylistic': tsStylistic,
+		},
+		rules: {
```

Comment:
the list here is quite exhaustive, i can imagine we relax a lot because of tests - if it's about that, please separate the rule overrides for tests and keep the actual code rules more strict